src/commands/run.rs

```diff
@@ -0,0 +1,263 @@
+use wdl::engine::v1::TaskEvaluator;
+
+use crate::Mode;
+use crate::analyze;
+use crate::get_display_config;
+
+/// Arguments for the run command.
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+pub struct RunArgs {
+    /// The path or URL to the WDL document containing the task to run.
+    #[arg(value_name = "PATH or URL")]
+    pub file: String,
+
+    /// The path to the input JSON file.
+    ///
+    /// If not provided, an empty set of inputs will be sent to the task.
+    #[arg(short, long, value_name = "JSON", conflicts_with = "name")]
+    pub inputs: Option<PathBuf>,
+
+    /// The name of the task to run.
+    ///
+    /// Required if no `inputs` file is provided.
+    #[arg(short, long, value_name = "NAME", conflicts_with = "inputs")]
+    pub name: Option<String>,
+
+    /// The output directory; defaults to the task name.
+    ///
+    /// If no output directory is provided, a default nested directory is
+    /// created based on the task name and the current time in the form
+    /// `sprocket_runs/<task_name>/<timestamp>/`.
+    #[arg(short, long, value_name = "DIR")]
+    pub output: Option<PathBuf>,
+
+    /// Overwrite the output directory if it exists.
+    #[arg(long)]
+    pub overwrite: bool,
+
+    /// Disables color output.
+    #[arg(long)]
+    pub no_color: bool,
+
+    /// The report mode.
+    #[arg(short = 'm', long, default_value_t, value_name = "MODE")]
+    pub report_mode: Mode,
+}
+
+/// Creates the output directory for the task.
+fn create_output_dir(output_dir: Option<PathBuf>, name: &str, overwrite: bool) -> Result<PathBuf> {
+    let output_dir = output_dir.unwrap_or_else(|| {
+        let timestamp = chrono::Utc::now().format("%Y%m%d_%H-%M-%S");
+        PathBuf::from(format!("sprocket_runs/{}/{}", name, timestamp))
+    });
+
+    let output_dir = if output_dir.exists() {
+        if overwrite {
+            fs::remove_dir_all(&output_dir).with_context(|| {
+                format!(
+                    "failed to remove output directory `{dir}`",
+                    dir = output_dir.display()
+                )
+            })?;
+            output_dir
+        } else {
+            bail!(
+                "output directory `{dir}` already exists; use the `--overwrite` option to \
+                 overwrite it",
+                dir = output_dir.display()
+            );
+        }
+    } else {
+        output_dir
+    };
+
+    fs::create_dir_all(&output_dir).with_context(|| {
+        format!(
+            "failed to create output directory `{dir}`",
+            dir = output_dir.display()
+        )
+    })?;
+
+    log::info!(
+        "output directory created: `{dir}`",
+        dir = output_dir.display()
+    );
+
+    Ok(output_dir)
+}
+
+/// Runs a task.
+pub async fn run(args: RunArgs) -> Result<()> {
+    eprintln!(
+        "the `run` command is in alpha testing and does not currently support workflows or using \
+         containers."
+    );
```

Comment:
With the warning being printed, I would suggest we use something like `workload`, `executable`, etc. instead of `task` to represent that this will eventually support larger executions.

Reply:
Ah, you mean in the doc comment? I was confused for a second there. Yea that makes sense to me, can update the comment.

Reply:
Actually, on second thought I think this should reflect the current capabilities. I don't think this comment should change. Same reasoning as explained higher up in another thread.